rocketpy/mathutils/function.py

```diff
@@ -1875,7 +1883,9 @@ def __add__(self, other):
                 return Function(lambda x: (self.get_value_opt(x) + other(x)))
         # If other is Float except...
         except AttributeError:
-            if isinstance(other, NUMERICAL_TYPES):
+            if isinstance(other, NUMERICAL_TYPES) or self.__is_single_element_array(
+                other
+            ):
```

Comment:
What was the old behavior that introduced the reason of this change?

To be clear, I am in favor of the change, just asking to understand the context.

Reply:
The problem is that we had `np.ndarray` in the `NUMERICAL_TYPES` list. This was problematic because we cannot say that `np.array([(0,0), (0, 1)])` is numeric. What we would like to say, actually, is: "numpy arrays can be considered numeric if they have only a single value".

I updated the NUMERICAL_TYPES but had to add a special comparison "is_single_element_array". 

I think the code is safer now.

Reply:
Agreed.